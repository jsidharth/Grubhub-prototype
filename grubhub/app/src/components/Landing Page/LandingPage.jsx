import React, { Component } from 'react';
import logo from './../../assets/grubhub-logo.jpg';
import { Link } from 'react-router-dom';

class Landingpage extends Component {
  render() {
    return (
      <div className="container">
        <div className="row">
          <div className="col-md front_logo">
            <img className="img-responsive" src={logo} alt="Grubhub"></img>
          </div>
          <div className="col-md text-center landing_button">
            <Link to="/signup">
              <button className="btn btn-outline-primary btn-lg m-5">Sign Up</button>
            </Link>
            <Link to="/signin">
              <button className="btn btn-outline-primary btn-lg">Sign In</button>
            </Link>
          </div>
        </div>
      </div>
    );
  }
}
export default Landingpage;
