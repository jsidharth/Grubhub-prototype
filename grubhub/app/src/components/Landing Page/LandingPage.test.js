// Link.react.test.js
import React from 'react';
import LandingPage from './LandingPage';
import renderer from 'react-test-renderer';

import { mount, shallow, render } from 'enzyme';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('Landing Page test', function() {

    it('should render correctly', () => {
      const component = shallow( <LandingPage/>);
      expect(component).toMatchSnapshot();
    });
    
    it('should be wrapped in a container div', () => {
        const component = shallow( <LandingPage/>);
        expect(component.find('.container').length).toEqual(1);
      });
    
      it('should have one Sign up button', () => {
        const component = shallow( <LandingPage/>);
        expect(component.find('.sign-up').length).toEqual(1);
      });
    
      it('should have one Sign in button', () => {
        const component = shallow( <LandingPage/>);
        expect(component.find('.sign-in').length).toEqual(1);
      });
});
