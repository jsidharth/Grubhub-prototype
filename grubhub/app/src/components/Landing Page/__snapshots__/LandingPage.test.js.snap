// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page test should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md front_logo"
    >
      <img
        alt="Grubhub"
        className="img-responsive"
        src="grubhub-logo.jpg"
      />
    </div>
    <div
      className="col-md text-center landing_button"
    >
      <Link
        to="/signup"
      >
        <button
          className="btn btn-outline-primary btn-lg m-5 sign-up"
        >
          Sign Up
        </button>
      </Link>
      <Link
        to="/signin"
      >
        <button
          className="btn btn-outline-primary btn-lg sign-in"
        >
          Sign In
        </button>
      </Link>
    </div>
  </div>
</div>
`;
